crates/bevy_ecs/src/system/mod.rs

```diff
@@ -564,6 +571,14 @@ mod tests {
         run_system(&mut world, sys);
     }
 
+    #[test]
+    #[should_panic = "&mut bevy_ecs::system::tests::A conflicts with a previous access in this query."]
+    fn any_of_with_conflicting() {
```

Comment:
Can you add tests for:

1. `&mut A` + `&A`
2.  `&A + &mut A` 
3. `&mut A` + Option<&A>`
4. `Entity` + `&mut A`
5. `() + &mut A`

as well? This is substantially trickier than I anticipated and I want to be sure all of our bases are covered as we refactor.

Reply:
makes sense! I added the unit tests